components/credentials/credentials-controller.js

```diff
@@ -20,6 +33,58 @@ function factory($scope, config, $location, ipCookie, localStorageService) {
   var trustedDomainPrefix = 'settings.trustedDomains.';
   var trustedDomainKey = trustedDomainPrefix + self.callbackHostName;
 
+  if(
+    self.identity.id !== self.requestOwner &&
+    self.identity.credential[0]['@graph'].claim.id !== self.requestOwner) {
+    brAlertService.add(
+      'error', 'Identity document does not correspond with the request.');
+    throw new Error('Identity document does not correspond with the request.');
+  }
+  var session = null;
+  var sessionKey = [self.identity.id.replace(/[:]/g,''), 'session'].join('.');
+  if(requestData.keyType === keyTypes.EXISTING) {
+    session = ipCookie(sessionKey);
+    if(session === undefined) {
+      // NOTE: stripping colon because browser encodes it
+      // this could be an acknowledgement where the owner is in claim.id
+      sessionKey = [self.identity.credential[0]['@graph']
+        .claim.id.replace(/[:]/g,''), 'session'].join('.');
```

Comment:
This session key stuff seems either hacky or brittle. What's going on? If we can't use standard encode/decode functions, then create helper encodeKey/decodeKey functions and call those.